internal/orchestrator/commiter.go

```diff
@@ -141,13 +143,19 @@ func (c *Commiter) commit(blockData []common.BlockData) error {
 
 	// TODO if next parts (saving or deleting) fail, we'll have to do a rollback
 	if err := c.saveDataToMainStorage(blockData); err != nil {
+		log.Error().Err(err).Msgf("Failed to commit blocks: %v", blockNumbers)
 		return fmt.Errorf("error saving data to main storage: %v", err)
 	}
 
 	if err := c.storage.StagingStorage.DeleteBlockData(blockData); err != nil {
+		log.Error().Err(err).Msgf("Failed to delete committed blocks from staging: %v", blockNumbers)
 		return fmt.Errorf("error deleting data from staging storage: %v", err)
```

Comment:
If you are logging here then there will be duplicate error logs, because `Start` will log the error coming from this method

Reply:
removed